File: trunk/quote/Projects/PluginModel/template/NoteProperties.vb
Imports System.ComponentModel

Namespace Template

    Public Class NoteProperties
        Inherits Common.NoteProperties

        Private _Note As String
        Private _Note2Customer As String

        Public Sub New()
    
        End Sub

        <DisplayName("Note-Internal"), _
         CategoryAttribute("Notes")> _
        Public Property Note As String
            Get
                Return _Note
            End Get
            Set(ByVal value As String)
                If _Note <> value Then
                    _Note = value
                    Me.SendEvents()
                End If
            End Set
        End Property

        <DisplayName("Note-To Customer"), _
         CategoryAttribute("Notes")> _
        Public Property Note2Customer As String
            Get
                Return _Note2Customer
            End Get
            Set(ByVal value As String)
                If _Note2Customer <> value Then
                    _Note2Customer = value
                    Me.SendEvents()
                End If
            End Set
        End Property

    End Class
End Namespace

File: trunk/quote/Projects/PluginModel/template/NoteProperties.cs
namespace Model.Template
{
    using System;
    using System.ComponentModel;
    using System.Linq;

    public sealed class NoteProperties : Common.NoteProperties
    {
        private string note;

        private string note2Customer;
        
        public NoteProperties()
        {
        }

        [DisplayName("Note-Internal"), CategoryAttribute("Notes")]
        public string Note 
        {
            get 
            { 
                return this.note; 
            }
            
            set 
            {
                if (this.note != value) 
                {
                    this.note = value;
                    this.OnPropertyChanged();
                }
            }
        }

        [DisplayName("Note-To Customer"), CategoryAttribute("Notes")]
        public string Note2Customer 
        {
            get 
            { 
                return this.note2Customer; 
            }
            
            set 
            {
                if (this.note2Customer != value) 
                {
                    this.note2Customer = value;
                    this.OnPropertyChanged();
                }
            }
        }
    }
}
